src/averageAges.js

```diff
@@ -37,11 +49,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const pAge = person => person.died - person.born;
+  const pSum = (x, y) => x + y;
+
+  if (withChildren === undefined) {
+    const allAge = people.filter(x => x.sex === 'f').map(pAge);
+
+    return Math.round(allAge.reduce(pSum) / allAge.length * 100) / 100;
+  } else {
+    const allAge = people.filter(x => x.name
+      && people.some(y => y.mother === x.name)).map(pAge);
+
+    return Math.round(allAge.reduce(pSum) / allAge.length * 100) / 100;
+  }
```

Comment:
Don't repeat yourself: return Math.round(allAge.reduce(pSum) / allAge.length * 100) / 100;
Use ternary operator for defining allAge array (allAge = withChildren ? : ) and then call reduce and return result only once